components/microsoft_onedrive/sources/common/utils.js

```diff
@@ -0,0 +1,24 @@
+module.exports = {
+  /**
+   * A utility function that accepts a string as an argument and reformats it in
+   * order to remove newline characters and consecutive spaces. Useful when
+   * dealing with very long templated strings that are split into multiple lines.
+   *
+   * @example
+   * // returns "This is a much cleaner string"
+   * toSingleLineString(`
+   *   This is a much
+   *   cleaner string
+   * `);
+   *
+   * @param {string}  multiLineString the input string to reformat
+   * @returns a formatted string based on the content of the input argument,
+   * without newlines and multiple spaces
+   */
+  toSingleLineString(multiLineString) {
+    return multiLineString
```

Comment:
Would be a good idea to wrap in String() so it doesn't break if `null` or `undefined` is passed.